lib/addons/helm-addon/index.ts

```diff
@@ -1,26 +1,66 @@
 import { Duration } from "aws-cdk-lib";
 import { Construct } from "constructs";
 import * as spi from "../..";
-import { cloneDeep } from "../../utils";
+import { cloneDeep, ConstraintsType, StringConstraint, UrlStringConstraint, validateConstraints } from "../../utils";
 import { HelmChartConfiguration, KubectlProvider } from "./kubectl-provider";
 
 export type HelmAddOnProps = HelmChartConfiguration;
 export type HelmAddOnUserProps = Partial<HelmChartConfiguration>;
 
+export class HelmAddonPropsConstraints implements ConstraintsType<HelmAddOnProps> {
+    /**
+    * chart can be no less than 1 character long, and no greater than 63 characters long.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+    */
+    chart = new StringConstraint(1, 63);
+
+    /**
+    * name can be no less than 1 character long, and no greater than 63 characters long.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+    */
+    name = new StringConstraint(1, 63);
+
+    /**
+    * namespace can be no less than 1 character long, and no greater than 63 characters long.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+    */
+    namespace = new StringConstraint(1, 63);
+
+    /**
+    * release can be no less than 1 character long, and no greater than 63 characters long.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+    */
+    release = new StringConstraint(1, 63);
+
+    /**
+    * repository can be no less than 1 character long, and no greater than 63 characters long. It also must follow a URL format.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
```

Comment:
This is wrong reference and max length is not 63. It can be a regular URL to a git repo. I suspect it could be at least 4KB long, however must follow a valid URL format.

Reply:
Would this be the correct reference?https://docs.aws.amazon.com/connect/latest/APIReference/API_UrlReference.html

Seems the name of the URL variable can be: Minimum length of 1. Maximum length of 4096.

While the URL itself can be: Length Constraints: Minimum length of 0. Maximum length of 4096.

Reply:
This one does not have to come from AWS site, it is a general URL format governed by RFC, e.g. https://datatracker.ietf.org/doc/html/rfc1738

Reply:
From my research there is no actual limit on it. But different browsers set different character limits. So that amazon link I believe should suffice since its the limits for amazon and being 4096 covers most things since lots of browsers limit to like 2000 something. Would you agree?